lib/node_modules/@stdlib/repl/lib/keybindings.js

```diff
@@ -16,334 +16,154 @@
-		}
-	],
-	'deleteLineRight': [
-		{
-			'name': 'delete',
-			'ctrl': true,
-			'shift': true,
-			'meta': false
-		},
-		{
-			'name': 'k',
-			'ctrl': true,
-			'shift': false,
-			'meta': false
-		}
-	],
-	'yankKilled': [
-		{
-			'name': 'y',
-			'ctrl': true,
-			'shift': false,
-			'meta': false
-		}
-	],
-	'yankPop': [
-		{
-			'name': 'y',
-			'ctrl': false,
-			'shift': false,
-			'meta': true
-		}
-	],
-	'undo': [
-		{
-			'name': '/',
-			'ctrl': true,
-			'shift': false,
-			'meta': false
-		},
-		{
-			'name': '-',
-			'ctrl': true,
-			'shift': false,
-			'meta': false
-		}
-	],
-	'redo': [
-		{
-			'name': '6',
-			'ctrl': true,
-			'shift': false,
-			'meta': false
-		},
-		{
-			'name': '6',
-			'ctrl': true,
-			'shift': true,
-			'meta': false
-		},
-		{
-			'name': '^',
-			'ctrl': true,
-			'shift': false,
-			'meta': false
-		},
-		{
-			'name': 'y',
-			'ctrl': false,
-			'shift': true,
-			'meta': false
-		}
-	],
-	'transposeAboutCursor': [
-		{
-			'name': 't',
-			'ctrl': true,
-			'shift': false,
-			'meta': false
-		}
-	],
-	'uppercaseNextWord': [
-		{
-			'name': 'u',
-			'ctrl': false,
-			'shift': false,
-			'meta': true
-		}
-	],
-	'capitalizeNextWord': [
-		{
-			'name': 'c',
-			'ctrl': false,
-			'shift': false,
-			'meta': true
-		}
-	],
-	'lowercaseNextWord': [
-		{
-			'name': 'l',
-			'ctrl': false,
-			'shift': false,
-			'meta': true
-		}
-	],
-	'clearScreen': [
-		{
-			'name': 'l',
-			'ctrl': true,
-			'shift': false,
-			'meta': false
-		}
-	]
-};
+function Keybindings( keybindings ) {
+	var actions;
+	var action;
+	var keys;
+	var key;
+	var i;
+	var j;
+
+	if ( !( this instanceof Keybindings ) ) {
+		return new Keybindings( keybindings );
+	}
+
+	// Initialize internal variables to store keybindings:
+	this._action2Keys = keybindings;
+	this._key2Action = {};
+
+	// Create a hash table of keybindings:
+	actions = objectKeys( keybindings );
+	for ( i = 0; i < actions.length; i++ ) {
+		action = actions[ i ];
+		keys = keybindings[ action ];
+		for ( j = 0; j < keys.length; j++ ) {
+			key = JSON.stringify( keys[ j ], REPLACER );
+			if ( hasOwnProp( this._key2Action, key ) ) {
+				throw new TypeError( format( 'invalid argument. Each key in the keybindings argument must correspond to a single action. Value: `%s`', key ) );
+			}
+			this._key2Action[ key ] = action;
+		}
+	}
+
+	return this;
+}
+
+/**
+* Sets a keybinding.
+*
+* @private
+* @name set
+* @memberof Keybindings.prototype
+* @type {Function}
+* @param {string} action - action name
+* @param {Array<Object>} keys - list of keys
+* @throws {TypeError} first argument must be a valid action name
+* @throws {TypeError} second argument must be an array of valid keys corresponding to a single action
+* @returns {void}
+*/
+setNonEnumerableReadOnly( Keybindings.prototype, 'set', function set( action, keys ) {
+	var validatedKeys;
+	var JSONKeys;
+	var key;
+	var obj;
+	var out;
+	var err;
+	var i;
+
+	if ( !contains( ACTIONS, action ) ) {
```

Comment:
@Snehil-Shah Use `@stdlib/array/base/assert/contains` factory here.

Reply:
In this case, `ACTIONS` is not coming from userland, so you can skip its validation. Hence, the recommendation to use `array/base/assert/contains`.